rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java

```diff
@@ -1117,6 +1127,43 @@ public Sha256Hash getBtcBlockchainBlockHashAtDepth(int depth) throws BlockStoreE
         return current.getHeader().getHash();
     }
 
+    /**
+     * @param btcTxHash The BTC transaction Hash
+     * @param btcBlockHash The BTC block hash
+     * @param merkleBranch The merkle branch
+     * @throws BlockStoreException
+     * @throws IOException
+     */
+    public Integer getBtcTransactionConfirmations(Sha256Hash btcTxHash, Sha256Hash btcBlockHash, MerkleBranch merkleBranch) throws BlockStoreException, IOException {
+        Context.propagate(btcContext);
+        this.ensureBtcBlockChain();
+
+        try {
+            // Get the block using the given block hash
+            StoredBlock block = btcBlockStore.getFromCache(btcBlockHash);
+            if (block == null) {
+                return BTC_TRANSACTION_CONFIRMATION_INEXISTENT_BLOCK_HASH_ERROR_CODE;
+            }
+
+            // Make sure it belongs to the best chain by diving to its height from the current best block
+            StoredBlock blockAtHeight = BridgeUtils.getStoredBlockAtHeight(this.btcBlockStore, block.getHeight());
+            if (!blockAtHeight.getHeader().getHash().equals(btcBlockHash)) {
+                return BTC_TRANSACTION_CONFIRMATION_BLOCK_NOT_IN_BEST_CHAIN_ERROR_CODE;
+            }
+
+            if (!merkleBranch.proves(btcTxHash, block.getHeader())) {
+                return BTC_TRANSACTION_CONFIRMATION_INVALID_MERKLE_BRANCH_ERROR_CODE;
+            }
+
+            return this.btcBlockChain.getBestChainHeight() - block.getHeight() + 1;
+        } catch (InvalidBlockHeightException | InvalidBlockHashException e) {
```

Comment:
InvalidBlockHeightException is used twice

Reply:
I believe one of them is `InvalidBlockHashException`

Reply:
true, sorry i miss read them